README.md

```diff
@@ -4,190 +4,126 @@ dns-gen sets up a container running Dnsmasq and [docker-gen].
+`172.17.42.1` but it could be something else. To known your IP, run the
+following command:
 
-    $ sudo systemctl status NetworkManager
+    GATEWAY=$(ip -4 addr show docker0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}')
+    echo ${GATEWAY}
 
-*step 2* Run dns-gen and bind port `53` to the `54`'s host
+Now, you can start the `dns-gen` container:
 
-    $ docker run --daemon --name dns-gen \
+    docker run -d --name dns-gen \
       --restart always \
-      --publish 54:53/udp \
+      --net host \
+      -e GATEWAY=$GATEWAY \
+      --log-opt "max-size=10m" \
+      --volume /:/host \
       --volume /var/run/docker.sock:/var/run/docker.sock \
-      jderusse/dns-gen -R
+      jderusse/dns-gen:2
 
-> the option `-R` just tell dns-gen to not fallback to the default resolver
-> which avoid an infinity loop of resolution
+You can test the container
 
+    docker run --name test nginx:alpine
+    dig test.docker @${GATEWAY}
 
-*step 3* Configure docker to use the `docker0` as DNS server
+Once OK, you can finally update your local resolver
 
-    # For systemd users (Fedora and recent Ubuntu versions) :
-    
-    $ vim /etc/systemd/system/docker.service
-    
-    # append
-    [Service]
-    ExecStart=
-    ExecStart=/usr/bin/docker daemon -H fd:// --bip=172.17.42.1/24 --dns=172.17.42.1
-    
-    # then
-    $ sudo systemctl daemon-reload
+    echo "nameserver ${GATEWAY}" | sudo tee --append /etc/resolvconf/resolv.conf.d/head
+    resolvconf -u
 
-    # For other users
-    $ vim /etc/default/docker
-    DOCKER_OPTS="--dns=172.17.42.1 --bip=172.17.42.1/24"
+**beware**! When your host will restart, it may change the IP address of
+the `docker0` interface. Mays may have to run the command `bin/install` to fix
+your configuration.
 
-    # In any cases
-    $ sudo service docker restart
+Or you can force docker to always use the same IP by editing the
+`/etc/docker/daemon.json` file and adding:
 
-Thank to this configuration the resolution workflow is now:
+    {
+      "bip": "172.17.42.1/24"
+    }
 
- * the host want to resolve `google.com`: `host` -> `dnsmasq` -> `external dns`
- * the host want to resolve `foo.docker`: `host` -> `dnsmasq` -> `127.0.0.1:54` -> `dns-gen`
- * a container want to resolve `google.com`: `container` -> `172.17.42.1` -> `dnsmasq` -> `external dns`
- * a container want to resolve `foo.docker`: `container` -> `172.17.42.1` -> `dnsmasq` -> `127.0.0.1:54` -> `dns-gen`
+**One more thing** When you start your host, the docker service is not fully
+loaded.
+Until this daemon is loaded, the dns container will not be automatically started
+and you will notice bad performance when your host will try to resolve DNS.
+The service is not fully loaded, because it uses a feature of systemd called
+[socket activation]: The first access to the docker socket will trigger the
+start of the true service.
+To skip this feature, you simply have to activate the docker service.
 
+    sudo update-rc.d docker enable
 
```

Comment:
## Uninstall

`docker update --restart=no dns-gen && docker stop dns-gen`

Reply:
I'm not sure to get your comment. I added a line in the section `Previous version of docker-gen`

Reply:
c'est pour desinstaller le container si on n'en pas plus besoin, juste un stop n'est pas suffisant